orchestrator/orchestrator_test.go

```diff
@@ -20,45 +17,121 @@ type TestSuite struct {
 	orchestrator *Orchestrator
 }
 
-func createTestSuite(t *testing.T) *TestSuite {
+type TestCase struct {
+	name    string
+	l1Host  string
+	l2Host  string
+	wantErr bool
+}
+
+func createTestSuite(t *testing.T, tc *TestCase) *TestSuite {
 	networkConfig := config.GetDefaultNetworkConfig(uint64(time.Now().Unix()), "")
 	testlog := testlog.Logger(t, log.LevelInfo)
 
+	l1Host := defaultHost
+	l2Host := defaultHost
+	if tc != nil {
+		l1Host = tc.l1Host
+		l2Host = tc.l2Host
+	}
+
+	networkConfig.L1Config.Host = l1Host
+	networkConfig.L1Config.Port = 9471
+	for i := range networkConfig.L2Configs {
+		networkConfig.L2Configs[i].Host = l2Host
+		networkConfig.L2StartingPort = 9473
+	}
+
 	ctx, closeApp := context.WithCancelCause(context.Background())
-	orchestrator, _ := NewOrchestrator(testlog, closeApp, &networkConfig)
+	orchestrator, err := NewOrchestrator(testlog, closeApp, &networkConfig)
+
 	t.Cleanup(func() {
 		closeApp(nil)
 		if err := orchestrator.Stop(context.Background()); err != nil {
 			t.Errorf("failed to stop orchestrator: %s", err)
 		}
 	})
 
+	if err != nil {
+		if tc != nil && tc.wantErr {
+			return nil
+		}
+		t.Fatalf("unable to create orchestrator: %s", err)
+	}
+
 	if err := orchestrator.Start(ctx); err != nil {
+		if tc != nil && tc.wantErr {
+			return nil
+		}
 		t.Fatalf("unable to start orchestrator: %s", err)
-		return nil
+	}
+
+	require.Equal(t, l1Host, orchestrator.L1Chain().Config().Host)
+	for _, chain := range orchestrator.l2Chains {
+		require.Equal(t, l2Host, chain.Config().Host)
 	}
 
 	return &TestSuite{t, orchestrator}
 }
 
 func TestStartup(t *testing.T) {
-	testSuite := createTestSuite(t)
+	testSuite := createTestSuite(t, nil)
+	verifyChainConfigs(t, testSuite)
+}
+
+func TestOrchestratorWithHosts(t *testing.T) {
+	t.Run("default configuration", func(t *testing.T) {
+		testSuite := createTestSuite(t, nil)
+		verifyChainConfigs(t, testSuite)
+	})
+	testCases := []TestCase{
+		{
+			name:    "custom hosts",
+			l1Host:  "0.0.0.0",
+			l2Host:  "0.0.0.0",
+			wantErr: false,
+		},
+		{
+			name:    "mixed custom hosts",
+			l1Host:  defaultHost,
+			l2Host:  "0.0.0.0",
+			wantErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Logf("Testing with L1 Host: %s, L2 Host: %s", tc.l1Host, tc.l2Host)
+
+			testSuite := createTestSuite(t, &tc)
+			if !tc.wantErr {
+				require.NotNil(t, testSuite)
+				verifyChainConfigs(t, testSuite)
+			}
+		})
+	}
+}
```

Comment:
the createTestSuite should ideally not have any tests in it.

If you want to test for the orchestrator failing to start I would add another test that runs

```
	orchestrator, err := NewOrchestrator(testlog, closeApp, &networkConfig)
```

and checks for error with out using createTestSuite